upgrader.go

```diff
@@ -119,10 +136,21 @@ func (u *Upgrader) Upgrade(ctx context.Context, t transport.Transport, maconn ma
 }
 
 func (u *Upgrader) setupSecurity(ctx context.Context, conn net.Conn, p peer.ID, dir network.Direction) (sec.SecureConn, bool, error) {
```

Comment:
Open question: Should the upgraded connection report the security protocol on `{Local, Remote}Multiaddr`? I tend to think that it should.